module/users/helper.js

```diff
@@ -898,47 +898,39 @@ module.exports = class UsersHelper {
   /**
    * Find non-targeted joined program.
    * @method
-   * @name getUserJoinedProgramDetailsWithPreviousProfiles
-   * @param {Array} targetedProgramIds - programIds
+   * @name getUserJoinedPrograms
    * @param {String} searchText - search text
    * @param {String} userId - userId
    * @returns {Object} - non-targeted joined program details.
    */
-  static getUserJoinedProgramDetailsWithPreviousProfiles(
-    targetedProgramIds,
-    searchText = "",
-    userId
-  ) {
+  static getUserJoinedPrograms(searchText = "", userId) {
     return new Promise(async (resolve, reject) => {
       try {
         let programUsersIds = [];
-        let nonTargettedProgramDetails = [];
+        let alreadyStartedPrograms = [];
 
         // find all programs joined by the user
         // programUsersData will contain list of programs joined by user from all the profiles. This can be considered as the super set of user programs
         let programUsersData = await programUsersHelper.programUsersDocuments(
           {
             userId: userId,
           },
-          ["programId"]
+          ["programId", "updatedAt"]
         );
 
         if (programUsersData.length > 0) {
+          //sort programs documents on updatedAt
+          programUsersData = programUsersData.sort(
```

Comment:
@ankitpws Can we make the DB query return in a sorted format so that we don't have to include the code here.

Reply:
@aks30 made changes